applications/luci-app-openwisp/htdocs/luci-static/resources/view/openwisp.js

```diff
@@ -0,0 +1,101 @@
+'use strict';
+'require view';
+'require form';
+'require tools.widgets as widgets';
+
+return view.extend({
+    render: function () {
+        var m, s, o;
+
+        m = new form.Map('openwisp',
+            _('OpenWISP'),
+            _("Configure, start and stop the OpenWISP agent on this device. Read more about configuration values: <a href='https://github.com/openwisp/openwisp-config'>https://github.com/openwisp/openwisp-config</a>"));
+
+        s = m.section(form.NamedSection, 'http', 'controller');
+
+        s.tab('general', _('General Settings'));
+        s.tab('advanced', _('Advanced Settings'));
+
+        // General settings
+        o = s.taboption('general', form.Value, 'uuid', _('UUID'), _('The UUID of this  device in the OpenWISP server.'))
+        o.readonly = true;
+
+        o = s.taboption('general', form.Value, 'key', _('Key'), _('The Key of this device in the OpenWISP server.'))
+        o.readonly = true;
+
+        o = s.taboption('general', form.ListValue, 'enabled',
+            _('Enable'),
+            _('Enable or disable the OpenWISP service.'));
+        o.value('disabled', _('Disabled'));
+        o.value('monitoring', _('Enabled'));
+
+        o = s.taboption('general', form.Value, 'url', _('Server URL'), _('The URL to the OpenWISP server. Example: https://openwisp2.mynetwork.com'))
+        o.placeholder = 'https://openwisp2.mynetwork.com'
+
+        o = s.taboption('general', form.Value, 'shared_secret', _('Shared Secret'), _('The organization shared secret for auto enrollment.'))
+        o.password = true
+
+        o = s.taboption('general', form.Value, 'interval', _('Update Interval'), _('How often to check in with the OpenWISP server. Expressed in seconds.'))
+        o.placeholder = '120'
+        o.datatype = 'uinteger';
+
+        // Advanced settings
+        o = s.taboption('advanced', form.Flag, 'verify_ssl', _('Verify SSL'), _('Verify the server SSL certificate.'))
+        o.rmempty = false
+
+        o = s.taboption('advanced', form.Flag, 'consistent_key', _('Consistent Key'), _('When using Automatic registration, this feature allows devices to keep the same configuration even if reset or reflashed.  This feature is enabled by default, but must be enabled also in the controller application in order to work.'))
+        o.rmempty = false
+
+        o = s.taboption('advanced', widgets.DeviceSelect, 'mac_interface', _('MAC Interface'), _('The interface to use for getting the MAC for this device.'))
+        o.rmempty = false
+
+        o = s.taboption('advanced', widgets.NetworkSelect, 'mac_interface', _('Management Interface'), _('The interface to use for management and control.'))
+        o.rmempty = false
```

Comment:
mac_interface is also used in the config option above?

Reply:
My mistake, fixed.